From ab0c3f2786e4d4248dc20c12a5b4ecad03127858 Mon Sep 17 00:00:00 2001
From: patchkez <patchkez@gmail.com>
Date: Sun, 14 Apr 2019 12:19:22 +0200
Subject: [PATCH] allow new lines in menu text

---
 consolemenu/menu_component.py | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/consolemenu/menu_component.py b/consolemenu/menu_component.py
index 09e0205..24e203f 100644
--- a/consolemenu/menu_component.py
+++ b/consolemenu/menu_component.py
@@ -247,8 +247,11 @@ class MenuTextSection(MenuComponent):
         for x in range(0, self.padding.top):
             yield self.row()
         if self.text is not None and self.text != '':
-            for line in textwrap.wrap(self.text, width=self.calculate_content_width()):
-                yield self.row(content=line, align=self.text_align)
+            list1 = self.text.split("\n")
+            for zz in list1:
+                for line in textwrap.wrap(str(zz), replace_whitespace=False,
+                        width=self.calculate_content_width()):
+                    yield self.row(content=line, align=self.text_align)
         for x in range(0, self.padding.bottom):
             yield self.row()
         if self.show_bottom_border:
-- 
2.21.0

